packages/abi-gen/test-cli/fixtures/contracts/AbiGenDummy.sol

```diff
@@ -138,4 +138,134 @@ contract AbiGenDummy
     uint someState;
     function nonPureMethod() public returns(uint) { return someState += 1; }
     function nonPureMethodThatReturnsNothing() public { someState += 1; }
+
+    /// @dev complex input is dynamic and more difficult to decode than simple input.
+    struct ComplexInput {
+        uint256 foo;
+        bytes bar;
+        string car;
+    }
+
+    /// @dev complex input is dynamic and more difficult to decode than simple input.
+    struct ComplexOutput {
+        ComplexInput input;
+        bytes lorem;
+        bytes ipsum;
+        string dolor;
+    }
+
+    /// @dev The fallback function calls into this contract and executes one of the above functions.
+    /// This allows us to test `getABIDecodedTransactionData` and `getABIDecodedReturnData` that is
```

Comment:
I'm not really sure how this can be used to test `getABIDecodedTransactionData()`. Where are the tests that use this?

Reply:
See below 🙏